Implemented model.xml parsing

c-mueller opened the pull request:
This PR implements the ability to parse a ``model.xml``. It puts the parsed data into the class structure defined in #6.

The implementation is not complete yet. The follownig aspects are still missing:
- [x] Parse nodes
- [x] Parse relationships
- [x] Connect nodes with the parsed relationships
- [x] Finish the specification
- [x] Fix the following issue with the parsing api: Currently documents containing spaces for better readability cannot be parsed properly. Currently all ``model.xml`` files need to be uglyfied (cannot contain spaces outside of elements)
- [ ] Document the code

**Important**: This Implementation builds on a slightly modified version of the class model described in #6 it is probably better to just merge this Pull request instead of #6.

Commit:
first experiments

c-mueller committed:
Added first class diagram draft

Still incomplete

c-mueller committed:
Added misc documentation

Added empty file for the TOSCAlite specification
Added example model.xml file to use for "inspiration"

c-mueller committed:
Merge remote-tracking branch 'origin/tosca2vsphere' into parser

c-mueller committed:
Updated class diagram

c-mueller committed:
Create java classes for object model

misc other changes

c-mueller committed:
Revert "Merge remote-tracking branch 'origin/tosca2vsphere' into parser"

This reverts commit c2f3c9dd026110ff07e1279b0e5cfdcf117ea9ff.

c-mueller committed:
Documentation on model classes

c-mueller committed:
Some Parser implementations

Commit:
continued to develop cli

mmunozba committed:
created dummy class with methods for engine

c-mueller committed:
Implemented some parsing features

The parser is now able to parse Node Properties and Machine nodes.

Commit:
Added import file handling

Commit:
Added the previous Executor

Not tested yet

Commit:
Missing .jar added

c-mueller committed:
Implemented artifact parsing

c-mueller committed:
Implemented relationship parsing and linking of nodes

c-mueller committed:
Improving codacy quality

c-mueller committed:
Specification and example Part 1

Commit:
added scripts for mysql

Commit:
Merging php-app and scripts in order to write scripts for database and php-app

hnicke commented:
what about changing the way how implementation artifacts are listed from 'nodename/scriptname' to 'scriptname'?
```xml
<Node>
<ImplemenationArtifacts>
<Start>start.sh</Start>
<-- instead of <Start>nodename/start.sh</Start> -->
</ImplementationArtifacts>
```

c-mueller commented:
Seems to be a good idea for Nodes, however it seems to make things more complicated when looking at ConnectsTo relationship artifacts.

hnicke commented:
If we establish this simple rule in the spec we can solve the ambiguity:
'ImplementationArtifacts of Relationsships have the static part of their relative path set to the source's directory.'
We could even offer both: preprend the path with / gives you absolute paths.

JensMueller2709 committed:
a first try for the creation method - create an additional class called creator

c-mueller committed:
Spec part 2

Commit:
Fixed bug that cli not terminated when file was not existent

Commit:
Added a guide for the cli

Commit:
changed cli.md for a better readability

c-mueller committed:
Spec part 3

Commit:
finished scripts for test application stack

Commit:
short unzip method added

tested it with Ubuntu, but command to check unzip package may not work on other distros

Commit:
added README for scripts dir; enhanced model.xml

Commit:
Added key-value mechanism to Deployment Artifacts

Commit:
renamed .zip archive into .tlite

Commit:
bugfix

c-mueller committed:
Merge branch 'parser' into engine

Commit:
Restructured Main Class

New Class ArchiveHandler to handle everything that has to do with the
TOSCAlite archive file

Commit:
Added tests and test resources for the ArchiveHandler

Commit:
Add zip files that were ignored because of the gitignore settings

JensMueller2709 committed:
correction of types and added reading of create properties in create method

Commit:
Added first test for the Main Class

c-mueller committed:
Implemented parsing capability for the reworked properties

c-mueller committed:
Fixed model.xml

The previous model.xml file did not parse, because of some missing node tags and a naming issue.

c-mueller committed:
Implemented relative artifact paths with the ArtifactPath class

c-mueller committed:
Externalized hard coded Element names into constants

c-mueller committed:
Implemented checks for Proper node names and property keys

c-mueller committed:
Some Documentation

Commit:
Merge remote-tracking branch 'origin/parser' into tosca2vsphere

Commit:
Merge remote-tracking branch 'origin/engine' into tosca2vsphere

Commit:
Removed example test

Commit:
Removed .DS_Store

Commit:
Removed .DS_Store

Commit:
Removed target

Commit:
Cleanup

nfode commented on src/main/java/de/toscana/transformator/model/ArtifactPath.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public class ArtifactPath {
+    private static final String PATH_SEPERATOR = "/";
+    private String absoultePath;
```
Typo: absolutePath

nfode commented on src/main/java/de/toscana/transformator/model/ArtifactPath.java:
```diff
@@ -0,0 +1,42 @@
+        absoultePath = PATH_SEPERATOR + "relationships" + PATH_SEPERATOR + path;
+    }
+
+    public String getAbsoultePath() {
```
Typo: getAbsolutePath

nfode commented on src/main/java/de/toscana/transformator/util/XMLUtils.java:
```diff
@@ -0,0 +1,7 @@
+package de.toscana.transformator.util;
```
Empty class?

nfode reviewed (commented):
Some Typos

Commit:
Added some comments

Commit:
Removed .idea folder and .iml files

c-mueller committed:
Fixed Typo

mklopp committed:
Merge branch 'master' into executor

c-mueller committed:
Added unit tests for some corner cases

Probably not adding more, there is just too many of them

mklopp committed:
added uploadAndUnzipZip method

c-mueller committed:
Specification draft v1

c-mueller committed:
Merge remote-tracking branch 'origin/master' into parser

c-mueller committed:
Removed iml file from git

JensMueller2709 committed:
improve create method - take care if node is machine or servicenode. minor bugfixes in class creator

nfode commented on doc/toscalite/spec.md:
```diff
@@ -0,0 +1,169 @@
+Implementation artifacts have to implement the lifecycle operations for a service. Due to TOSCAlite requiring Linux-based systems shell scripts have to be used to perform these operations.
+
+The following operations are supported:
+* Create - Required by every service. Used to create the service. This has to install all the packages required to run the service, for example.
```
This has to... Weird sentence. I might come up with a better later this day. Atm i am to tired.

nfode commented on doc/toscalite/spec.md:
```diff
@@ -0,0 +1,169 @@
+
+### Machines
+
+A machine node represents a Linux-based system, probably Ubuntu 16.04. The deployment system does not care where the system is running, as long as the machine is reachable with SSH.
```
Maybe stress that ssh is mandatory! 
I think it might be better to say the machine is reachable and ssh is installed and properly configured.
Because without ssh nothing will work :smile:

nfode reviewed (commented):
Some minor proposals.

mmunozba committed:
updated comments

mmunozba committed:
create a class to test the engine

mmunozba committed:
Merge branch 'parser' into engine

mmunozba committed:
created test topology

mmunozba committed:
.idea folder and .iml files updated to solve testing issues

Commit:
Merged tosca2vshere with executor

Commit:
integrated scripts branch

mmunozba committed:
updated .gitignore

mmunozba committed:
removed .idea folder

Commit:
integrated engine

Commit:
removed  target file from git

c-mueller committed:
Removed unwanted files

c-mueller committed:
Fixed complaints found by @nfode in the specification

c-mueller committed:
Updated example.md

Commit:
fixed model.xml, commented out some stuff so we can build project

c-mueller committed:
Merge remote-tracking branch 'origin/master' into parser

Commit:
added how to get and how to use guide to readme

Commit:
fixed typo

Commit:
updated .gitignore

Commit:
index on integration: c3a4d4f updated .gitignore

Commit:
WIP on integration: c3a4d4f updated .gitignore

Commit:
Removed unnecessary misc files

Commit:
Merge branch 'tosca2vsphere-doc' into integration

Commit:
clean up

JensMueller2709 committed:
creator should now be able to read topology with more than one vm

JensMueller2709 committed:
major bugfixing - wrong logic so far

JensMueller2709 committed:
update test

JensMueller2709 committed:
write some tests and fixed a bug in creator

JensMueller2709 committed:
start method is now available

mmunozba committed:
Merge branch 'executor' into engine

mmunozba committed:
removed .idea and iml from merge with executor

mmunozba committed:
attempt n2 @ removing *.iml and .idea from executor merge

mmunozba committed:
updated .gitignore again

JensMueller2709 committed:
implement stop method

mmunozba committed:
SSH connection enabled

mmunozba committed:
send SHH commands implemented (not fully functional)

Commit:
removed intellij files

mmunozba committed:
edited comments, implemented ZIP-Upload, removed exampleClass

mmunozba committed:
updated comments

mmunozba committed:
removed typo

Commit:
Added sudo to unzip installation

Sudo requiered to install unzip on a normal Ubuntu machine

Commit:
Sleep is not necessary

Commit:
Added Runner

mklopp committed:
changed filenames strings to files

mklopp committed:
Merge branch 'executor' of https://github.com/nfode/stupro_toscana_vorprojekt into executor

mklopp committed:
fixed

Commit:
removed Example classes

Commit:
Merge branch 'executor' of github.com:nfode/stupro_toscana_vorprojekt into executor

mklopp committed:
added update and upgrade

mklopp committed:
updated travis icon

mklopp committed:
Update README.md

mklopp committed:
overloaded function with nodename

JensMueller2709 committed:
update Readme

JensMueller2709 committed:
change ssh command - added nodename

mklopp committed:
added a function which first cd's into folder

c-mueller committed:
Replaced System.out calls in the parser with logging

c-mueller committed:
Updated specification

Commit:
small correction

Commit:
small corrections

mklopp committed:
changed executer methods and added execute script function

mklopp committed:
executor now uploads util scipts and puts them into place

Commit:
model.xml's machines now point to our vsphere vms; make sure to use vpn and add your ssh key before usage.

Commit:
Merge branch 'scripts' into integration

Commit:
merged master into branch integration

Commit:
merged executor into integration

Commit:
merged the updated parser into integration branch

Commit:
merged engine udpates into integration branch

Commit:
added zip

Commit:
Merge remote-tracking branch 'origin/engine' into tosca2vsphere

Commit:
Merge remote-tracking branch 'origin/executor' into tosca2vsphere

Commit:
Excluded executor tests from mvn test

Commit:
Merge remote-tracking branch 'origin/executor' into engine

Commit:
Merge branch 'engine' of github.com:nfode/stupro_toscana_vorprojekt into engine

Commit:
Clean up

Commit:
fixed merge; project now compiles

Commit:
removed line from gitignore

Commit:
Update integration with the latest code from 'tosca2vsphere'

Commit:
Merge branch 'integration' of github.com:nfode/stupro_toscana_vorprojekt into integration

Commit:
removed g

Commit:
added automatism to create simple-app.zip

Commit:
changed String param to file

Commit:
Merge branch 'tosca2vsphere' into integration

Commit:
Fixed typo in model.xml

Commit:
fixed create-zip script

Commit:
removed naming mistakes

Commit:
removed duplicated model.xml

Commit:
fixed model.xml

Commit:
merged scripts branch into integration branch

JensMueller2709 committed:
Merge remote-tracking branch 'origin/executor' into engine

 the commit.
new methods in executor

JensMueller2709 committed:
update ssh method cause of changing method in executor

Commit:
modified engine and executor; reworked error propagation; cleaned up executor interface

Commit:
added logger config file; fixed file seperator bug in executor

Commit:
bugfix: executor.executeScript() now treats the incoming path correctly

Commit:
cli can now be exited with command 'exit'

Commit:
improved logging; bugfixed model.xml; removed zip, without machine passwords files must be packed manually

Commit:
added cli option

Commit:
Merge branch 'engine' into integration

JensMueller2709 committed:
change craete method - nodes will be started in same moment now

Commit:
executor now prints vm's standard out to console

Commit:
Merge branch 'integration' into engine

JensMueller2709 committed:
check if artifacttype is null to avoid nullpointer exception

JensMueller2709 committed:
Merge branch 'engine' of https://github.com/nfode/stupro_toscana_vorprojekt into engine

JensMueller2709 committed:
execute connects to relationships

JensMueller2709 committed:
bug fixes

JensMueller2709 committed:
bug fixes

mklopp committed:
now getting both stdout and stderr

mklopp committed:
Merge branch 'executor' into integration

Commit:
Merge branch 'engine' into integration

mklopp committed:
minor fixes

Commit:
Merge branch 'executor' into integration

mklopp committed:
updated executer with evironment variables

mklopp committed:
Merge branch 'executor' into integration

JensMueller2709 committed:
bug fixes

Commit:
Merge branch 'engine' of github.com:nfode/stupro_toscana_vorprojekt into engine

Commit:
Merge branch 'engine' into integration

Commit:
Changed executor interface

mklopp committed:
hash map to map and envars into executeScript

Commit:
Fixed logger and added status to prompt

Commit:
removed all unnecessary log libs from pom

Commit:
Removed buggy test

Commit:
Fixed test

c-mueller committed:
Merge branch 'executor' into integration

Commit:
Again fixed executors executeScript bug. keep attention while merging!

Commit:
improved logging and command printing of sendCommand

Commit:
refactored executor; improved logging

Commit:
added shutdown hook which closes ssh connections

Commit:
bugfixed executors close() method
